mini-macro/src/lib.rs

```diff
@@ -7,6 +7,9 @@ extern crate proc_macro;
 use proc_macro::{quote, TokenStream};
 
 #[proc_macro_derive(ClippyMiniMacroTest)]
+/// # Panics
+///
+/// Panics if the macro derivation fails
```

Comment:
This should not be necessary.  The panic happens in an external macro ``quote!`` which will be confusing to the user, if they get a warning from this lint.

Reply:
That makes sense. Does that mean we need to change something in FindPanicUnwrap to not check if we are in a macro?

Reply:
Well we need to check somehow for macros in this visitor. The problem here is, that `in_external_macro` won't really work here, because ``panic`` and so on are also external macros (I think). It would be great, if you could try experimenting with `utils::is_expn_of` and `is_direct_expn_of` and then only lint when we're inside a ``std`` panic macro (``panic``/``unreachable``). Also maybe not ``todo`` or ``unimplemented``, since those macros signal that the implementation is unfinished and panics may not occur in the finished function. 

So rather than checking just for ``match_panic_def_id``, you can try to also check if the expansion is coming from ``panic`` or ``unreachable``.

Reply:
Actually, after thinking it over some more, I think I respectfully disagree @flip1995 . Users should be aware of macros that panic within their function and document that. If we need to make the warning clearer to show that the macro panics, or point to the place in the macro where it panics, that's okay.

Reply:
That is a good point. And since this lint is `pedantic` I think this is reasonable. I'm convinced :+1: